src/PID_TEST.cc

```diff
@@ -121,7 +121,14 @@ TEST(PidTest, EqualOperatorCornerCase)
   EXPECT_DOUBLE_EQ(pid.CmdOffset(), 1.23);
   EXPECT_DOUBLE_EQ(pid.Cmd(), 0.0);
 
+#ifndef _WIN32
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wself-assign-overloaded"
```

Comment:
I'm getting a warning about this: `warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]` on gcc 9.3. Maybe it's only recognized in clang?

Reply:
Looking at it again, I believe this whole test is irrelevant since we switch to the `ImplPtr` pattern.  I have removed it.